From: Benjamin Drung <benjamin.drung@canonical.com>
Date: Wed, 11 Feb 2026 15:29:54 +0100
Subject: fix(syslog): use return instead of exit in initqueue/online hook

The `rsyslogd-start.sh` initqueue/online hook calls `exit 0`. This
causes the sourcing script to exit. The subsequent code is not executed.

Use `return` instead of `exit` in the syslog initqueue/online hook.

Fixes: https://github.com/dracut-ng/dracut-ng/issues/2227
Forwarded: https://github.com/dracut-ng/dracut-ng/pull/2238
---
 modules.d/77syslog/rsyslogd-start.sh | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/modules.d/77syslog/rsyslogd-start.sh b/modules.d/77syslog/rsyslogd-start.sh
index 9682a8b..92ce925 100755
--- a/modules.d/77syslog/rsyslogd-start.sh
+++ b/modules.d/77syslog/rsyslogd-start.sh
@@ -7,7 +7,7 @@ command -v getarg > /dev/null || . /lib/dracut-lib.sh
 # prevent starting again if already running
 if [ -f /var/run/syslogd.pid ]; then
     read -r pid < /var/run/syslogd.pid
-    kill -0 "$pid" && exit 0
+    kill -0 "$pid" && return 0
 fi
 
 rsyslog_config() {
